app/models/casebook/resource.rb

```diff
@@ -0,0 +1,32 @@
+# == Schema Information
+#
+# Table name: casebooks
+#
+#  id            :integer          not null, primary key
+#  title         :string           default("Untitled casebook"), not null
+#  slug          :string
+#  subtitle      :string
+#  headnote      :text
+#  public        :boolean          default(TRUE), not null
+#  root_id       :integer
+#  ordinals      :integer          default([]), not null, is an Array
+#  copy_of_id    :integer
+#  is_alias      :boolean
+#  material_type :string
+#  material_id   :integer
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#
+
+class Casebook::Resource < Casebook::Contents
+  default_scope {where.not(material_id: nil)}
+
+  belongs_to :material, polymorphic: true, inverse_of: :casebooks, required: true
```

Comment:
Yeah, there's a naming issue. "Casebook" is not totally right for this concept.

Think of the abstract class `Casebook::Generic` as any node in a tree representation of a Table of Contents. It could be the root (a Casebook), a branch (a Section), or a leaf (a Resource).

All three share a conceptual model in that they have a position in a Table of Contents, they have a `title`, `headnote`, `collaborators`, they can be public or not, they can be cloned, and so on.

They differ in that only Casebooks don't have a root (or have themselves as root), and only Resources have a reference to a piece of real content (a Case or TextBlock or Link).

So yes: a Resource is a reference to a case which shares behavior with an abstract class of objects that includes Sections and Casebooks. In the data model, that makes it a row in `casebooks`; but that's not a good name, because a Resource is conceptually _not_ the same thing as a Casebook.

We could call the abstract module/table something more generic like ContentNode?